### PR TITLE
Implemented a simple MySQL DSN wrapper. First step towards fixing #227

### PR DESCRIPTION
**Pull request by tobihille**
The current Version still has some issues, e.g. using "initstatements" still leads to an exception so it does not work for magento yet. I will try to fix this tomorrow if I can.

I assume it is not the nicest code because im quite unfamiliar with C# but even if you reject the PR you should get an idea what i wanted to fix, so I think it helps in both cases ;-)

### CODE REVIEW

**Comment by bfistein** [comment]
Your contributions are much appreciated @tobihille. We first have to fix our previous commit, and we'll review your PR very soon after. Thanks again.

**Comment by jakubmisek** [comment]
something like that will be needed, ideally if `MySqlConnector` would understand `dbname` directly.

File: src/PDO/Peachpie.Library.PDO/PDOMySQLWrapper.cs [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+                return dsn;
+            }
+
+            dsn = dsn.Replace("host", "Host");
+            dsn = dsn.Replace("port", "Port");
+            dsn = dsn.Replace("dbname", "Database");
+            //TODO: if unix_socket is provided the protocol probably needs to be set as well.
+            // but scince I have no way of testing this I do not touch it.
+            //dsn = dsn.Replace("unix_socket", "..."); 
+            dsn = dsn.Replace("charset", "CharSet");
```
Comment (jakubmisek) [inline src/PDO/Peachpie.Library.PDO/PDOMySQLWrapper.cs]: casing should not make difference

File: src/PDO/Peachpie.Library.PDO/PDOMySQLWrapper.cs [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+        public static string convertDSN(string dsn)
+        {
+            if (dsn == null || dsn.Length == 0)
+            {
+                return dsn;
+            }
+
+            dsn = dsn.Replace("host", "Host");
+            dsn = dsn.Replace("port", "Port");
+            dsn = dsn.Replace("dbname", "Database");
```
Comment (jakubmisek) [inline src/PDO/Peachpie.Library.PDO/PDOMySQLWrapper.cs]: we have the idea, of course we cannot change "dbname" to "Database" in connection string, that would be an unpleasant sideeffect.

**Comment by jakubmisek** [comment]
looking at the code, we should parse the DNS by ourselves in order to create valid connection options

**Comment by jakubmisek** [comment]
Tobi, I'll close the PR and try to implement it more in general. Thanks, if you'd have any other hints please feel free to create another PR.

**Pull request closed without merging**